src/seml/commands/add.py

```diff
@@ -301,30 +370,35 @@ def add_config_file(
         seml_config['working_dir'],
     )
 
+    # Upload source files: This also determines the batch_id
+    if seml_config['use_uploaded_sources'] and not no_code_checkpoint:
+        seml_config['source_files'] = upload_sources(seml_config, collection, batch_id)
+    del seml_config['use_uploaded_sources']  # type: ignore
+    # Now it is in the right format for the type checks
+    seml_config = cast(SemlDoc, seml_config)
```

Comment:
I added new types in document.py, which represent the file content. Though these are very rudimentary, I recommend fleshing them out later instead of growing this PR.

Reply:
Isn't this the kind of the point of the PR? Why do they need to be flashed out, or put differently, why does `read_config` not comply with the proposed `SemlDoc` already?

Reply:
because the config in the yaml file differs from the config in the db, several keys only exist in either, I have added new types to capture this. But one PR doesn't have to finish the whole typing thing.